src/fetchToMongoTask.js

```diff
@@ -37,11 +34,35 @@ class FetchToMongoTask extends cronious.Task {
 
     run (progressCallback) {
         return this._provider(progressCallback)
-            .then(definitions => this._collection.findOneAndUpdate(
-                { _id: this._documentId },
-                definitions,
-                { upsert: true }
-            ));
+            .then((definitions) => {
+
+                const changeId = new mongodb.ObjectId();
+                const updates = Object.keys(definitions).map((key) => {
+                    const newDocument = {
+                        _id: key,
+                        changeId,
+                        type: 'features',
+                        definitions: definitions[key]
+                    };
+
+                    return {
+                        updateOne: {
+                            filter: { _id: key },
+                            update: newDocument,
+                            upsert: true
+                        }
+                    };
+                });
+
+                return this._collection.bulkWrite(updates)
```

Comment:
Nice-to-have: je nutné, aby update operace proběhli striktně uspořádaně? Pokud ne, využil bych možnost nastavit ordered: false a umožnit tak databázi zvýšit výkonnost při operaci. Viz. poznámky v dokumentaci https://docs.mongodb.com/v3.2/reference/method/db.collection.bulkWrite/